std/experimental/allocator/package.d

```diff
@@ -1123,33 +1123,85 @@ allocated with the same allocator.
 */
 void dispose(A, T)(auto ref A alloc, T* p)
 {
+    if (!p) return;
     static if (hasElaborateDestructor!T)
+        p.__dtor;
+    alloc.deallocate((cast(void*)p)[0 .. T.sizeof]);
+}
+
+/// Ditto
+void dispose(A, T)(auto ref A alloc, T p)
+if (is(T == class) && T.stringof != Object.stringof)
+{
+    if (p)
     {
-        destroy(*p);
+        auto support = (cast(void*) p)[0 .. typeid(p).initializer.length];
+        static if (__traits(hasMember, T, "__dtor"))
+        {
+            p.__dtor;
```

Comment:
Will this free the assosiated object monitor? See [this](https://github.com/dlang/druntime/blob/v2.071.0/src/rt/lifetime.d#L1352) for reference.
I also think that it would be easiest if you forward to `rt_finalize` directly for the dynamic case base.